src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2012-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "wallet/coinselection.h"
+#include "util.h"
+#include "utilmoneystr.h"
+
+// Descending order comparator
+struct {
+    bool operator()(CInputCoin a, CInputCoin b) const
+    {
+        return a.txout.nValue > b.txout.nValue;
+    }
+} descending;
+
+struct CompareValueOnly
+{
+    bool operator()(const CInputCoin& t1,
+                    const CInputCoin& t2) const
+    {
+        return t1.txout.nValue < t2.txout.nValue;
+    }
+};
+
+bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_value, const CAmount& cost_of_change, std::set<CInputCoin>& out_set,
+    CAmount& value_ret, std::vector<CAmount>& input_fee_vec, std::vector<CAmount>& long_term_fee_vec, CAmount& fee_ret)
+{
+    out_set.clear();
+    value_ret = 0;
+
+    if (utxo_pool.empty()) {
+        return false;
+    }
+
+    int depth = 0;
+    int remaining_tries = 100000;
+    std::vector<std::pair<bool, bool>> selection; // First bool: select the utxo at this index; Second bool: traversing exclusion branch of this utxo
+    selection.assign(utxo_pool.size(), std::pair<bool, bool>(false, false));
+    bool done = false;
+    bool backtrack = false;
+
+    // Sort the utxo_pool
+    std::sort(utxo_pool.begin(), utxo_pool.end(), descending);
```

Comment:
Don't you need to update `input_fee_vec` and `long_term_fee_vec` during the sort? Maybe it would be simpler to make `input_fee` and `long_term_fee` new fields in `CInputCoin`.

Reply:
Yeah, I do... I took your suggestion and added fields to `CInputCoin` for that coin's `fee` and `long_term_fee` values.